api_app/views.py

```diff
@@ -417,6 +423,82 @@ def analyze_multiple_observables(request):
     )
 
 
+@add_docs(
+    description="""
+    REST endpoint to fetch list of job comments or
+    retrieve/delete a job comment with job comment ID.
+    Requires authentication.
+    """
+)
+class CommentViewSet(ReadAndDeleteOnlyViewSet, CreateModelMixin, SerializerActionMixin):
+    queryset = Comment.objects.all()
+    serializer_class = CommentSerializer
+    serializer_action_classes = {
+        "list": CommentListSerializer,
+        "retrieve": CommentSerializer,
+        "create": CommentCreateSerializer,
+    }
+
+    # check if the user is the owner of the
+    # job or if the user is a member of the same org
+    class IsCommentObjectOwnerOrSameOrgPermission(BasePermission):
+        def has_object_permission(self, request, view, obj):
+            if request.user.has_membership():
+                if (
+                    request.user.membership.organization
+                    == obj.job.user.membership.organization
+                ):
+                    return True
+
+            if request.user == obj.job:
+                return True
+            return False
+
+    class IsCommentObjectOwner(BasePermission):
+        def has_object_permission(self, request, view, obj):
+            if request.user == obj.user:
+                return True
+            return False
+
+    def get_permissions(self):
+        permissions = super().get_permissions()
+
+        # only the owner of the comment can update or delete the comment
+        if self.action in ["destroy", "update", "partial_update"]:
+            permissions.append(self.IsCommentObjectOwner())
+
+        # anyone can read the comment
```

Comment:
what? no only comment owner can.

Reply:
mine was only a suggestion to change the comment, I should have been more explicit in that sense.

Reply:
is it fine if we leave it the way it is? i will edit it in the next PR if you want me to. i want to finish this issue safely before we get into the visualizer hassle.

Reply:
i am not sure if i like this logic right now. i think only organization admins should be able to do all that rather than anyone in the organization, to begin with. this is why i would like to wait out the change.